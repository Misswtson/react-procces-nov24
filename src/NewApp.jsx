import PropTypes from 'prop-types';

export const NewApp = ( { title, subTitle, name } ) => {


    return (
<>
  <h1 data-testid="test-tittle"> { title } </h1>
  {/* comand shift slach  */} 
  {/* <code> {JSON.stringify( NewMessage)}<code/>*/}
    <p>{ subTitle }</p>
    <p>{ name }</p>

    </>
    );
}

NewApp.propTypes = {
title: PropTypes.string.isRequired,
subTitle: PropTypes.number
}

NewApp.defaultProps = {
    name:'Emma',
    title:'Hola, soy Emma',
    subTitle: 'Lo lograras, ya lo ver√†s!!'
}