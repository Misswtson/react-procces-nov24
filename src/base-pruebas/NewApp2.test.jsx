import { getByText, render, screen } from '@testing-library/react';
import { NewApp } from '../NewApp';


describe('pruebas en NewApp.jsx', () => {

    const tittle = 'Hola, soy Emma';
    const subTitle = 'Lo lograras, ya lo verÃ s!!'

   test('Debe hacer Match con el Sanpshot', () => { 
    const { container } = render(<NewApp tittle={ tittle } />);  
    expect( container ).toMatchSnapshot();
})

test('Debe mostrar el mensaje "Hola, soy Emma"', () => { 

    render( <NewApp tittle={ tittle }/>);
    expect (screen.debug());

 })

 test('Debe mostrar el titulo en un h1', () => { 
    render(<NewApp tittle={ tittle } />);
    expect( screen.getByRole('heading', { level: 1 }).innerHTML).toContain( tittle );
  })

  test('Debe mostrar el subtitulo enviado por props', () => { 
        
  render( 
    < NewApp 
        tittle = { tittle }
        subTitle ={ subTitle }
        /> 
        );
    expect( screen.getAllByText(subTitle).length ).toBe(1);
})
});
