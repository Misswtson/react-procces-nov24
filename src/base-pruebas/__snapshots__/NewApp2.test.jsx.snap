// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en NewApp.jsx Debe hacer Match con el Sanpshot 1`] = `
<div>
  <h1
    data-testid="test-tittle"
  >
     
    Hola, soy Emma
     
  </h1>
  <p>
    Lo lograras, ya lo ver√†s!!
  </p>
  <p>
    Emma
  </p>
</div>
`;
